### ISSUE TITLE
Adicionar Categoria

### ISSUE THREAD

**Issue by dearrudam** [arrudalabs/simple-finance-backend#10]
**Necessidades:**

De acordo com a especificação, implementar um endpoint para possibilitar o cadastramento de categorias onde:
- cada categoria tem uma descrição;
- cada categoria deve ser do tipo "CREDIT" ou "DEBIT";

E criar teste de contrato para validar o comportamento. 

**Restrições:**
- descrição é requerido;
- tipo é requerido;
- não deve permitir duplicação de categorias com o mesmo tipo e descrição

 **Resultado esperado**
- retornar 201 com a categoria  cadastrada e um header "Location" com a URL  para acessar a nova categoria
- Em caso de violação das restrições, retorne 400 com as mensagens de erro

**Dicas**:  
*"***Disclamer:*** essas dicas são para estudar e conhecer um pouco mais sobre o Spring e sobre os frameworks envolvidos, então não encarem como uma regra, não existe "bala de prata", mas uma ou outra "ferramenta" a mais em nosso cinto de utilidades não faz mal algum, não é ?"* 
- Para as validações, que tal utilizar a especificacao Bean Validation para validar o input da sua requisição? Tente criar validadores para as restrições que a especificação exige. 
- Menos é mais: não tente recriar ou adicionar alguma solução pensando em casos futuros que até então não existem. O software vai evoluir e a naturalmente sua complexidade também. 
- Entre em contato com o grupo do projeto em caso de dúvida. Que tal fazer pair programming? Ou quem sabe sessões de coding dojo?

Segue a especificação OpenAPI v3 para esse endpoint: [clique aqui](https://github.com/arrudalabs/simple-finance-backend/blob/main/src/main/resources/static/openapi.yaml)

Uma versão reduzida:
```yaml
paths:
  /api/categories:
    post:
      description: Add a new category
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/NewCategory'
      responses:
        201:
          description: Retrieves the created category
          headers:
            Location:
              description: the relative URL of the new category
              schema:
                type: string
                format: url
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Category'
        default:
          description: Invalid request
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
components:
  schemas:
    NewCategory:
      type: object
      properties:
        type:
          type: string
          format: enum
          enum: ['CREDIT','DEBIT']
        description:
          type: string
      
    Category:
      type: object
      properties:
        id:
          type: number
        type:
          type: string
          format: enum
          enum: ['CREDIT','DEBIT']
        description:
          type: string
         
    Errors:
      type: object
      properties:
        mensagens:
          type: array
          items:
            type: string

```

**Comment by rmarinheiro** [comment]
Sugestão: Que tal colocarmos uma paginação nesse endpoint? 

Eu sei que não está na especificação da API, mas eu acredito que além de ajudar o front no momento da listagem de categorias, também é uma forma de aprender um recurso útil.

Outro detalhe que reparei(me corrijam se eu estiver falando besteira), mas nesse endpoint não precisaríamos ter alguma referencia pro tipo Usuário? Isto é não seria necessário identificar a que usuário pertence determinada categoria?

**Comment by dearrudam** [comment]
Sim... podemos melhorar essa API assim como adicionar um padrão para paginação em nossa API
